src/server/array_func.c

```diff
@@ -569,23 +569,20 @@ get_subjob_state(job *parent, int iindx)
 void
 update_subjob_state_ct(job *pjob)
 {
-	char *buf;
+	char buf[BUF_SIZE];
 	static char *statename[] = {
```

Comment:
Hey, can you make this `const` as well? When it gets converted to c++, the compiler will complain.